### ISSUE TITLE
Official example: Scan List is Empty

### ISSUE THREAD

**Issue by Tyrn** [Polidea/FlutterBleLib#37]
Hi,

Scan list is empty. `BluetoothAdapter(22120): 616386226: getState(). Returning 12` in console probably means that everything is OK. I can see one bluetooth device in my Android > Settings. Permissions for the example app on the Android device look OK. Android OS is 5.1.

It all looks exactly the same with the `flutter_blue` example, by the way. I suspect I should clear my target device settings somehow, but I am at a loss. Any ideas?

**Comment by luqmanhakem** [comment]
After using this example, I thought my BLE beacon was the problem. But after I tried several other apps, I do notice that it does work well.

Tried in debugging mode, and it returned this error message as I tap the search button.

> Launching lib\main.dart on CLT L29 in debug mode...
Initializing gradle...
Resolving dependencies...
Running 'gradlew assembleDebug'...
Note: H:\flutterproject\FlutterBleLib\example\build\flutter_ble_lib\generated\source\proto\debug\javalite\com\polidea\flutterblelib\BleData.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
Built build\app\outputs\apk\debug\app-debug.apk.
Installing build\app\outputs\apk\app.apk...
I/FlutterActivityDelegate(23249): onResume setting current activity to this
Syncing files to device CLT L29...
I/zygote64(23249): Do partial code cache collection, code=28KB, data=26KB
I/zygote64(23249): After code cache collection, code=28KB, data=26KB
I/zygote64(23249): Increasing code cache capacity to 128KB
V/AudioManager(23249): playSoundEffect   effectType: 0
E/MethodChannel#flutter_ble_lib(23249): Failed to handle method call
E/MethodChannel#flutter_ble_lib(23249): rx.exceptions.OnErrorFailedException: Error occurred when trying to propagate error to Observer.onError
E/MethodChannel#flutter_ble_lib(23249): 	at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:187)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:115)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.exceptions.Exceptions.throwOrReport(Exceptions.java:212)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:48)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:35)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.Observable.subscribe(Observable.java:10352)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.Observable.subscribe(Observable.java:10319)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.Observable.subscribe(Observable.java:10159)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.BleHelper.safeStartDeviceScan(BleHelper.java:144)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.BleHelper.startDeviceScan(BleHelper.java:134)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.FlutterBleLibPlugin.startDeviceScan(FlutterBleLibPlugin.java:241)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.FlutterBleLibPlugin.onMethodCall(FlutterBleLibPlugin.java:97)
E/MethodChannel#flutter_ble_lib(23249): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:191)
E/MethodChannel#flutter_ble_lib(23249): 	at io.flutter.view.FlutterNativeView.handlePlatformMessage(FlutterNativeView.java:152)
E/MethodChannel#flutter_ble_lib(23249): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#flutter_ble_lib(23249): 	at android.os.MessageQueue.next(MessageQueue.java:379)
E/MethodChannel#flutter_ble_lib(23249): 	at android.os.Looper.loop(Looper.java:144)
E/MethodChannel#flutter_ble_lib(23249): 	at android.app.ActivityThread.main(ActivityThread.java:7377)
E/MethodChannel#flutter_ble_lib(23249): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#flutter_ble_lib(23249): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:469)
E/MethodChannel#flutter_ble_lib(23249): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:963)
E/MethodChannel#flutter_ble_lib(23249): Caused by: rx.exceptions.CompositeException: 2 exceptions occurred. 
E/MethodChannel#flutter_ble_lib(23249): 	... 21 more
E/MethodChannel#flutter_ble_lib(23249): Caused by: rx.exceptions.CompositeException$CompositeExceptionCausalChain: Chain of Causes for CompositeException In Order Received =>
E/MethodChannel#flutter_ble_lib(23249): 	at android.util.Log.printlns(Log.java:442)
E/MethodChannel#flutter_ble_lib(23249): 	at android.util.Log.e(Log.java:274)
E/MethodChannel#flutter_ble_lib(23249): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:208)
E/MethodChannel#flutter_ble_lib(23249): 	... 8 more
E/MethodChannel#flutter_ble_lib(23249): Caused by: com.polidea.rxandroidble.exceptions.BleScanException: Location Permission missing (code 3)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.rxandroidble.internal.scan.ScanPreconditionsVerifierApi18.verify(ScanPreconditionsVerifierApi18.java:29)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.rxandroidble.internal.scan.ScanPreconditionsVerifierApi24.verify(ScanPreconditionsVerifierApi24.java:38)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.rxandroidble.RxBleClientImpl$1.call(RxBleClientImpl.java:114)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.rxandroidble.RxBleClientImpl$1.call(RxBleClientImpl.java:111)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:46)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:35)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.Observable.subscribe(Observable.java:10352)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.Observable.subscribe(Observable.java:10319)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.Observable.subscribe(Observable.java:10159)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.BleHelper.safeStartDeviceScan(BleHelper.java:144)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.BleHelper.startDeviceScan(BleHelper.java:134)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.FlutterBleLibPlugin.startDeviceScan(FlutterBleLibPlugin.java:241)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.FlutterBleLibPlugin.onMethodCall(FlutterBleLibPlugin.java:97)
E/MethodChannel#flutter_ble_lib(23249): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:191)
E/MethodChannel#flutter_ble_lib(23249): 	... 8 more
E/MethodChannel#flutter_ble_lib(23249): Caused by: java.lang.IllegalArgumentException: Unsupported value: com.polidea.rxandroidble.exceptions.BleScanException: Location Permission missing (code 3)
E/MethodChannel#flutter_ble_lib(23249): 	at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:293)
E/MethodChannel#flutter_ble_lib(23249): 	at io.flutter.plugin.common.StandardMethodCodec.encodeErrorEnvelope(StandardMethodCodec.java:70)
E/MethodChannel#flutter_ble_lib(23249): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.error(MethodChannel.java:199)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.FlutterBleLibPlugin$DefaultOnErrorAction.onError(FlutterBleLibPlugin.java:449)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.BleHelper$2.call(BleHelper.java:152)
E/MethodChannel#flutter_ble_lib(23249): 	at com.polidea.flutterblelib.BleHelper$2.call(BleHelper.java:149)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.internal.util.ActionSubscriber.onError(ActionSubscriber.java:44)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:153)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:115)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.exceptions.Exceptions.throwOrReport(Exceptions.java:212)
E/MethodChannel#flutter_ble_lib(23249): 	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:48)
E/MethodChannel#flutter_ble_lib(23249): 	... 17 more
V/AudioManager(23249): playSoundEffect   effectType: 0
E/flutter (23249): [ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
E/flutter (23249): PlatformException(error, Error occurred when trying to propagate error to Observer.onError, null)
E/flutter (23249): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:547:7)
E/flutter (23249): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:279:18)
E/flutter (23249): <asynchronous suspension>
E/flutter (23249): #2      FlutterBleLib.startDeviceScan (file:///H:/flutterproject/FlutterBleLib/lib/source/lib_core.dart:108:30)
E/flutter (23249): <asynchronous suspension>
E/flutter (23249): #3      BleDevicesState._onStartScan (package:flutter_ble_lib_example/ui/ble_devices_screen.dart:252:10)
E/flutter (23249): #4      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:494:14)
E/flutter (23249): #5      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:549:30)
E/flutter (23249): #6      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:102:24)
E/flutter (23249): #7      TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:161:9)
E/flutter (23249): #8      TapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:94:7)
E/flutter (23249): #9      PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:315:9)
E/flutter (23249): #10     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:73:12)
E/flutter (23249): #11     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:101:11)
E/flutter (23249): #12     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:143:19)
E/flutter (23249): #13     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:121:22)
E/flutter (23249): #14     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:101:7)
E/flutter (23249): #15     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:64:7)
E/flutter (23249): #16     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:48:7)
E/flutter (23249): #17     _invoke1 (dart:ui/hooks.dart:134:13)
E/flutter (23249): #18     _dispatchPointerDataPacket (dart:ui/hooks.dart:91:5)

>

**Comment by Tyrn** [comment]
Can't see any errors. Does it depend on log level or something?

**Comment by faridishere** [comment]
Hi 

I am experience the same problem. The list is empty and scanning exits immediately with error. 

D/ViewRootImpl@7f6e1cd[MainActivity](30031): ViewPostIme pointer 0
D/ViewRootImpl@7f6e1cd[MainActivity](30031): ViewPostIme pointer 1
E/MethodChannel#flutter_ble_lib(30031): Failed to handle method call
E/MethodChannel#flutter_ble_lib(30031): rx.exceptions.OnErrorFailedException: Error occurred when trying to propagate error to Observer.onError
E/MethodChannel#flutter_ble_lib(30031): 	at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:187)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:115)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.exceptions.Exceptions.throwOrReport(Exceptions.java:212)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:48)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:35)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.Observable.subscribe(Observable.java:10352)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.Observable.subscribe(Observable.java:10319)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.Observable.subscribe(Observable.java:10159)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.BleHelper.safeStartDeviceScan(BleHelper.java:144)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.BleHelper.startDeviceScan(BleHelper.java:134)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.FlutterBleLibPlugin.startDeviceScan(FlutterBleLibPlugin.java:241)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.FlutterBleLibPlugin.onMethodCall(FlutterBleLibPlugin.java:97)
E/MethodChannel#flutter_ble_lib(30031): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:191)
E/MethodChannel#flutter_ble_lib(30031): 	at io.flutter.view.FlutterNativeView.handlePlatformMessage(FlutterNativeView.java:152)
E/MethodChannel#flutter_ble_lib(30031): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#flutter_ble_lib(30031): 	at android.os.MessageQueue.next(MessageQueue.java:325)
E/MethodChannel#flutter_ble_lib(30031): 	at android.os.Looper.loop(Looper.java:142)
E/MethodChannel#flutter_ble_lib(30031): 	at android.app.ActivityThread.main(ActivityThread.java:6944)
E/MethodChannel#flutter_ble_lib(30031): 	at java.lang.reflect.Method.invoke(Native Method)
E/MethodChannel#flutter_ble_lib(30031): 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
E/MethodChannel#flutter_ble_lib(30031): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
E/MethodChannel#flutter_ble_lib(30031): Caused by: rx.exceptions.CompositeException: 2 exceptions occurred. 
E/MethodChannel#flutter_ble_lib(30031): 	... 21 more
E/MethodChannel#flutter_ble_lib(30031): Caused by: rx.exceptions.CompositeException$CompositeExceptionCausalChain: Chain of Causes for CompositeException In Order Received =>
E/MethodChannel#flutter_ble_lib(30031): 	at android.util.Log.printlns(Log.java:539)
E/MethodChannel#flutter_ble_lib(30031): 	at android.util.Log.e(Log.java:331)
E/MethodChannel#flutter_ble_lib(30031): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:208)
E/MethodChannel#flutter_ble_lib(30031): 	... 8 more
E/MethodChannel#flutter_ble_lib(30031): Caused by: com.polidea.rxandroidble.exceptions.BleScanException: Location Permission missing (code 3)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.rxandroidble.internal.scan.ScanPreconditionsVerifierApi18.verify(ScanPreconditionsVerifierApi18.java:29)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.rxandroidble.internal.scan.ScanPreconditionsVerifierApi24.verify(ScanPreconditionsVerifierApi24.java:38)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.rxandroidble.RxBleClientImpl$1.call(RxBleClientImpl.java:114)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.rxandroidble.RxBleClientImpl$1.call(RxBleClientImpl.java:111)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:46)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:35)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.Observable.subscribe(Observable.java:10352)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.Observable.subscribe(Observable.java:10319)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.Observable.subscribe(Observable.java:10159)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.BleHelper.safeStartDeviceScan(BleHelper.java:144)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.BleHelper.startDeviceScan(BleHelper.java:134)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.FlutterBleLibPlugin.startDeviceScan(FlutterBleLibPlugin.java:241)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.FlutterBleLibPlugin.onMethodCall(FlutterBleLibPlugin.java:97)
E/MethodChannel#flutter_ble_lib(30031): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:191)
E/MethodChannel#flutter_ble_lib(30031): 	... 8 more
E/MethodChannel#flutter_ble_lib(30031): Caused by: java.lang.IllegalArgumentException: Unsupported value: com.polidea.rxandroidble.exceptions.BleScanException: Location Permission missing (code 3)
E/MethodChannel#flutter_ble_lib(30031): 	at io.flutter.plugin.common.StandardMessageCodec.writeValue(StandardMessageCodec.java:293)
E/MethodChannel#flutter_ble_lib(30031): 	at io.flutter.plugin.common.StandardMethodCodec.encodeErrorEnvelope(StandardMethodCodec.java:70)
E/MethodChannel#flutter_ble_lib(30031): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler$1.error(MethodChannel.java:199)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.FlutterBleLibPlugin$DefaultOnErrorAction.onError(FlutterBleLibPlugin.java:449)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.BleHelper$2.call(BleHelper.java:152)
E/MethodChannel#flutter_ble_lib(30031): 	at com.polidea.flutterblelib.BleHelper$2.call(BleHelper.java:149)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.internal.util.ActionSubscriber.onError(ActionSubscriber.java:44)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:153)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:115)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.exceptions.Exceptions.throwOrReport(Exceptions.java:212)
E/MethodChannel#flutter_ble_lib(30031): 	at rx.internal.operators.OnSubscribeDefer.call(OnSubscribeDefer.java:48)
E/MethodChannel#flutter_ble_lib(30031): 	... 17 more
E/flutter (30031): [ERROR:topaz/lib/tonic/logging/dart_error.cc(16)] Unhandled exception:
E/flutter (30031): PlatformException(error, Error occurred when trying to propagate error to Observer.onError, null)
E/flutter (30031): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:547:7)
E/flutter (30031): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:279:18)
E/flutter (30031): <asynchronous suspension>
E/flutter (30031): #2      FlutterBleLib.startDeviceScan (file:///C:/Users/MOBASSEF/AndroidStudioProjects/ble_lib/lib/source/lib_core.dart:108:30)
E/flutter (30031): <asynchronous suspension>
E/flutter (30031): #3      BleDevicesState._onStartScan (package:flutter_ble_lib_example/ui/ble_devices_screen.dart:252:10)
E/flutter (30031): #4      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:494:14)
E/flutter (30031): #5      _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:549:30)
E/flutter (30031): #6      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:102:24)
E/flutter (30031): #7      TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:161:9)
E/flutter (30031): #8      TapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:94:7)
E/flutter (30031): #9      PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:315:9)
E/flutter (30031): #10     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:73:12)
E/flutter (30031): #11     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:101:11)
E/flutter (30031): #12     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:143:19)
E/flutter (30031): #13     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:121:22)
E/flutter (30031): #14     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:101:7)
E/flutter (30031): #15     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:64:7)
E/flutter (30031): #16     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:48:7)
E/flutter (30031): #17     _invoke1 (dart:ui/hooks.dart:134:13)
E/flutter (30031): #18     _dispatchPointerDataPacket (dart:ui/hooks.dart:91:5)

**Comment by JamesMcIntosh** [comment]
```
 Caused by: java.lang.IllegalArgumentException: Unsupported value: com.polidea.rxandroidble.exceptions.BleScanException: Location Permission missing (code 3)
```

You need to request the `Location` in the app for BlueTooth to work.

For more info see: https://github.com/Polidea/RxAndroidBle#observing-client-state

**Comment by pawelByszewski** [comment]
@Tyrn @JamesMcIntosh @faridingithub @luqmanhakem It has been a long time since we stopped maintaining the library but now we come with great news! We are currently working with Frontside on a new BLE Flutter library and BLE simulator. Beta version has been released already and we are going to release 2.0.0 in the first half of November
